### PR TITLE
Create menu component

### LINKED ISSUE #237: Create menu component and profile button
### Description
Create general menu

### Requirements

- general menu must have slot for any child as source
- must be responsive
- must be dark-mode ready

### Deliverables and requirements

- `MenuSlim` - simple menu with links
- `ProfileButton` - profile button with image

### Reference

image

### PR DESCRIPTION
**Pull request by Abdurasulovich**
Closes #237

### CODE REVIEW

File: src/Training.AirBnb.Clone.Frontend/src/Modules/profile/components/Darkmode.vue [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<template>
+  <div :class="modelValue ? 'theme-bg-primary-toggle' : 'bg-gray-300'" @click="toggle" class="w-10 h-6 flex primary-transition-toggle theme-bg-accent rounded-full p-1 theme-text-primary">
+    <div  :class="modelValue ? 'translate-x-4' : 'translate-x-0'" class="w-4 h-4 primary-transition-toggle bg-textSecondary rounded-full theme-shadow"></div>
+  </div>
```
Comment (Ilxom-22) [inline src/Training.AirBnb.Clone.Frontend/src/Modules/profile/components/Darkmode.vue:4]: Toggle ga transition berib qo'ying. Silliq harakatlanishi uchun.

File: src/Training.AirBnb.Clone.Frontend/src/Modules/profile/components/ProfileMenuComponent.vue [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      <a href="http://localhost:5173/">Gift cards</a>
+    </button>
+
+    <button class="text-left theme-text-primary font-sans font-light p-3 w-full theme-hover-shadow">
+      <a href="http://localhost:5173/">Airbnb your home</a>
+    </button>
+
+    <button class="text-left theme-text-primary font-sans font-light p-3 w-full theme-hover-shadow">
+      <a href="http://localhost:5173/">Help Center</a>
+    </button>
```
Comment (Ilxom-22) [inline src/Training.AirBnb.Clone.Frontend/src/Modules/profile/components/ProfileMenuComponent.vue:23]: Buttonlar hover bo'lganda background rangi o'zgarmayabti.

**Review by Ilxom-22 — CHANGES_REQUESTED** [review]

File: src/Training.AirBnb.Clone.Frontend/src/Modules/profile/components/Darkmode.vue [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<template>
+  <div :class="modelValue ? 'theme-bg-primary-toggle' : 'bg-gray-300'" @click="toggle" class="w-10 h-6 flex primary-transition-toggle theme-bg-accent rounded-full p-1 theme-text-primary">
+    <div  :class="modelValue ? 'translate-x-4' : 'translate-x-0'" class="w-4 h-4 primary-transition-toggle bg-textSecondary rounded-full theme-shadow"></div>
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Frontend/src/Modules/profile/components/Darkmode.vue:3]: - rename this component to `DarkModeToggler`
- remove toggle value as it uses `AppThemeService` service internally

**Review by sulton-max — COMMENTED** [review]
Good job

File: src/Training.AirBnb.Clone.Frontend/src/Modules/profile/components/ProfileMenuComponent.vue [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+const appThemeService = new AppThemeService();
+
+const toggleState = ref(appThemeService.isDarkMode() ? true : false!);
+
+const props = defineProps({
+  value:{
+    type: Boolean
+  }
+})
+const emit = defineEmits(['update:value', 'onClose']);
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Frontend/src/Modules/profile/components/ProfileMenuComponent.vue:47]: Remove props and update emit, as this is not input component

**Review by sulton-max — CHANGES_REQUESTED** [review]

**Review by Abdurasulovich — COMMENTED** [review]
Fine

**Review by Abdurasulovich — COMMENTED** [review]
Done

**Review by sulton-max — APPROVED** [review]
Good job

**Review by Ilxom-22 — APPROVED** [review]

**Review by sulton-max — APPROVED** [review]
Well done

**Pull request merged**